frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/SimpleGraphTable.java

```diff
@@ -10,33 +9,30 @@
 import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositorySnapshot;
 import com.virtuslab.gitmachete.backend.api.NullGitMacheteRepositorySnapshot;
 import com.virtuslab.gitmachete.frontend.graph.api.repository.IRepositoryGraphCache;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseGraphTable;
 import com.virtuslab.gitmachete.frontend.ui.impl.cell.BranchOrCommitCell;
 import com.virtuslab.gitmachete.frontend.ui.impl.cell.BranchOrCommitCellRendererComponent;
 
-public final class DemoGraphTable extends JBTable implements IGitMacheteRepositorySnapshotProvider {
-
-  private static final GraphTableModel GRAPH_TABLE_MODEL = deriveGraphTableModel();
-
+public final class SimpleGraphTable extends BaseGraphTable implements IGitMacheteRepositorySnapshotProvider {
   @Getter
   private final IGitMacheteRepositorySnapshot gitMacheteRepositorySnapshot = NullGitMacheteRepositorySnapshot.getInstance();
 
   @UIEffect
-  public static DemoGraphTable deriveInstance() {
+  public static SimpleGraphTable deriveInstance(IGitMacheteRepositorySnapshot macheteRepositorySnapshot) {
     // We can keep the data - graph table model,
     // but wee need to reinstantiate the UI - demo graph table.
-    return new DemoGraphTable(GRAPH_TABLE_MODEL);
+    return new SimpleGraphTable(deriveGraphTableModel(macheteRepositorySnapshot));
   }
 
   @UIEffect
-  private static GraphTableModel deriveGraphTableModel() {
+  private static GraphTableModel deriveGraphTableModel(IGitMacheteRepositorySnapshot macheteRepositorySnapshot) {
     var repositoryGraphCache = RuntimeBinding.instantiateSoleImplementingClass(IRepositoryGraphCache.class);
-    var gitMacheteRepositorySnapshot = new DemoGitMacheteRepositorySnapshot();
-    var repositoryGraph = repositoryGraphCache.getRepositoryGraph(gitMacheteRepositorySnapshot, /* isListingCommits */ true);
+    var repositoryGraph = repositoryGraphCache.getRepositoryGraph(macheteRepositorySnapshot, /* isListingCommits */ true);
     return new GraphTableModel(repositoryGraph);
   }
 
   @UIEffect
-  private DemoGraphTable(GraphTableModel graphTableModel) {
+  private SimpleGraphTable(GraphTableModel graphTableModel) {
     super(graphTableModel);
```

Comment:
Nit: while you're here, reorganize the `SimpleGraphTable` c'tor so that it's as similar to `EnhancedGraphTable` c'tor as possible (explicit `setColumnSelectionAllowed` etc. would be nice here as well)

Reply:
As in, they're already pretty aligned:
![image](https://user-images.githubusercontent.com/3383210/89343560-bfdaf580-d6a4-11ea-8fce-cdd07f0bf491.png)